mp/src/game/client/momentum/ui/HUD/hud_speedometer_label.cpp

```diff
@@ -65,29 +65,52 @@ void SpeedometerLabel::ApplySchemeSettings(IScheme *pScheme)
     m_IncreaseColor = GetSchemeColor("MOM.Speedometer.Increase", pScheme);
     m_DecreaseColor = GetSchemeColor("MOM.Speedometer.Decrease", pScheme);
     m_pComparisonLabel->SetFont(GetFont());
+    m_pComparisonLabel->SetPos(GetXPos(), GetYPos());
+    m_pComparisonLabel->SetWide(GetWide());
+    m_pComparisonLabel->SetContentAlignment(GetContentAlignment());
     Reset();
 }
 
 void SpeedometerLabel::PerformLayout()
 {
     BaseClass::PerformLayout();
 
-    // keep centered in parent
+    if (m_eColorizeType != SPEEDOMETER_COLORIZE_COMPARISON_SEPARATE || !m_bDrawComparison)
+        return;
+
     char szMain[BUFSIZELOCL];
     GetText(szMain, BUFSIZELOCL);
+    char szComparison[BUFSIZELOCL];
+    m_pComparisonLabel->GetText(szComparison, BUFSIZELOCL);
 
     HFont labelFont = GetFont();
-    int combinedLength = UTIL_ComputeStringWidth(labelFont, szMain);
+    int iMainTextLength = UTIL_ComputeStringWidth(labelFont, szMain);
+    int iCompTextLength = UTIL_ComputeStringWidth(labelFont, szComparison);
+    int iCombinedLength = iMainTextLength + iCompTextLength;
 
-    if (m_eColorizeType == SPEEDOMETER_COLORIZE_COMPARISON_SEPARATE && m_bDrawComparison)
+    switch (GetContentAlignment())
     {
-        char szComparison[BUFSIZELOCL];
-        m_pComparisonLabel->GetText(szComparison, BUFSIZELOCL);
-        combinedLength += UTIL_ComputeStringWidth(labelFont, szComparison);
+        // left
+        case a_northwest:
+        case a_west:
+        case a_southwest:
+            SetTextInset(0, 0);
+            m_pComparisonLabel->SetTextInset(iMainTextLength, 0);
+            break;
+        case a_center:
+            SetTextInset((iMainTextLength - iCombinedLength) / 2, 0);
+            m_pComparisonLabel->SetTextInset((iCompTextLength - iCombinedLength) / 2 + iMainTextLength, 0);
```

Comment:
Doesn't `iMainTextLength - iCombinedLength` == `iMainTextLength - (iMainTextLength + iCompTextLength)` == `iCompTextLength` ? Same for `iCompTextLength - iCombinedLength`.